src/components/general/AuthContext.js

```diff
@@ -67,7 +91,34 @@ function AuthProvider({ children, setProfile, setTasks, setPets }) {
       const userTasks = allTasks.filter((task) => {
         return task.userId === user
       })
-      setTasks(userTasks)
+
+      db.collection('users').onSnapshot((snapshot) => {
+        const allUsers = snapshot.docs.map((doc) => ({
+          id: doc.id,
```

Comment:
Are you always fetching for all users all data and then filtering in the react app?
You should change the Firebase queries, so that the data is filtered in database and the react app has only the access to data it should have. this way I could easily YOUR data (if I had an account), which shouldnt be the case from a security point of view

Reply:
I suspected that would be a problem. My logic for showing pets to users was 
1. the user.id and pet.userId are the same 
and/or
2. if the user's email is included in someone else's user document under the family array, they can see their pets too
I'm not sure how I can specify that as Firebase security rules.